nipype/interfaces/ants/preprocess.py

```diff
@@ -0,0 +1,305 @@
+            return "frame_displacement.csv"
+        return None
+
+    def _list_outputs(self):
+        outputs = self._outputs().get()
+        if _extant(self.inputs.output_spatial_map):
+            outputs['spatial_map'] = (
+                os.path.abspath(self.inputs.output_spatial_map)
+            )
+        if _extant(self.inputs.output):
+            outputs['output'] = os.path.abspath(self.inputs.output)
+        return outputs
+
+class AntsMotionCorrInputSpec(ANTSCommandInputSpec):
+    '''Input spec for the antsMotionCorr command.'''
+    dimension_desc = (
+        "This option forces the image to be treated as a "
+        "specified-dimensional image. If not specified, N4 tries to infer "
+        "the dimensionality from the input image."
+    )
+    dimensionality = traits.Enum(3, 2, argstr='-d %d', usedefault=True,
+                                 position=0, desc=dimension_desc, default=3)
+
+    average_image = File(argstr='-a %s', position=1, exists=False,
+                         desc="Average the input time series image.")
+
+    output_average_image = File(hash_files=False, desc="", argstr="%s",
+                                genfile=True, exists=False, usedefault=True)
+    output_transform_prefix = traits.Str()
+    output_warped_image = File(hash_files=False, desc="",
+                               exists=False)
+
+    metric_type = traits.Enum("CC", "MeanSquares", "Demons", "GC", "MI",
+                              "Mattes", argstr="%s")
+    fixed_image = File(requires=['metric_type'], desc="")
+    moving_image = File(requires=['metric_type'],
+                        desc="This is the 4d image to be motion corrected")
+    metric_weight = traits.Float(1.0, requires=['metric_type'])
+    radius_or_bins = traits.Int(desc="", requires=['metric_type'])
+    sampling_strategy = traits.Enum("None", "Regular", "Random", None,
+                                    requires=['metric_type'])
+    sampling_percentage = traits.Either(traits.Range(low=0.0, high=1.0), None,
+                                        requires=['metric_type'])
+
+    transformation_model = traits.Enum("Affine", "Rigid", argstr="%s")
+    gradient_step_length = traits.Float(requires=['transformation_model'],
+                                        desc='')
+
+    iterations = traits.Int(
+        argstr="-i %d",
+        desc="Specify the number of iterations at each level."
+    )
+    smoothing_sigmas = traits.Int(
+        argstr="-s %d",
+        desc="Specify the amount of smoothing at each level."
+    )
+    shrink_factors = traits.Int(
+        argstr="-f %d",
+        desc=("Specify the shrink factor for the virtual domain (typically "
+              "the fixed image) at each level.")
+    )
+    n_images = traits.Int(
+        argstr="-n %d",
+        desc=("This option sets the number of images to use to construct the "
+              "template image.")
+    )
+
+    use_fixed_reference_image = traits.Bool(
```

Comment:
- does ```-u``` need a value? I think this is just a switch (```argstr='-u'```).
- Please, set the default value using positional arguments (```traits.Bool(True, argstr=...)```)